### PR TITLE
Move knativeteam-groups Prow jobs to a separate file and delete unused backup Prow job

### PR DESCRIPTION
**Pull request by chizhg**
/cc @upodroid @kvmware

### CODE REVIEW

File: prow/jobs/custom/test-infra.yaml [hunk 1]
```diff
@@ -154,32 +135,6 @@ periodics:
         requests:
           cpu: 100m
           memory: 1Gi
-- cron: "15 9 * * *"
-  name: ci-knative-backup-artifacts
```
Comment (kvmware) [inline prow/jobs/custom/test-infra.yaml:158]: Was it intentional to remove this job?

Comment (kvmware) [reply prow/jobs/custom/test-infra.yaml:158]: ah, seems like unit tests caught this was missing. If it is an intentional removal then looks like unit tests will need to be updated as well.

Comment (chizhg) [reply prow/jobs/custom/test-infra.yaml:158]: Yes, it has been failing for a long while - https://prow.knative.dev/?job=ci-knative-backup-artifacts

**Review by upodroid — COMMENTED** [review]
👍

**Pull request merged**